runtime/router.go

```diff
@@ -160,7 +166,15 @@ func NewRouterEndpoint(
 func (endpoint *RouterEndpoint) HandleRequest(
 	w http.ResponseWriter, r *http.Request, params httprouter.Params,
 ) {
+	reqFields := logRequestFields(r)
+	resFields := make([]zapcore.Field, 0)
+
+	defer func() {
+		writeLogs(endpoint.gateway.Logger, reqFields, resFields)
```

Comment:
No, if you defer it will pass the current arrays of resFields as an argument. This is reassigned later to a different underlying array later in this method. The anonymous closure will reference the new underlying array.